### DISCUSSION TITLE
automated wcc installtion

### DISCUSSION THREAD

**Discussion by Ultimecia1463** [open-watcom/open-watcom-v2#1346] [Q&A] [answered]
```
RUN echo '#!/bin/bash\n' \
             'wget https://github.com/open-watcom/open-watcom-v2/releases/download/Current-build/open-watcom-2_0-c-linux-x64 -O /tmp/open-watcom-installer\n' \
              'chmod +x /tmp/open-watcom-installer\n' \
              '/tmp/open-watcom-installer -s --install-dir=/usr/bin/watcom\n' \ 
              'rm /tmp/open-watcom-installer\n' \
```

i wanted to installed wcc in docker but this is not working i cant find wcc in /usr/bin/watcom/binl

can someone help indentify what flag to pass for non-interactive install of 16-bit compiler

**Comment by jmalak** [comment, marked as answer]
We have a project https://github.com/open-watcom/setup-watcom to install Open Watcom for use with CI build, but it is not what you are searching.
Generaly more suitable for automation is to use OW binary snapshot instead of installer image even if both contains same image. Installers are suitable for interactive installation and snapshot for automatic installation.
snapshot is available for any build that no problem with availability the download image for Current-build is https://github.com/open-watcom/open-watcom-v2/releases/download/Current-build/owsnapshot.tar.xz.
it is simple tar archive that contains also executable attribute setup if it is used on *NIX systems.
simply restore it to your preffered location and setup appropriate environment variable WATCOM, PATH, INCLUDE for your host and target system. If your host system is Linux and you want to compile for DOS target then you can setup these wariables following way.

```
export WATCOM=<location of OW root>
export PATH=$WATCOM/binl64:$WATCOM/binw:$PATH
export INCLUDE=$WATCOM/h
```

**Reply by Ultimecia1463** [reply]
are there any examples similar to my use case
im noob pls help

**Reply by jmalak** [reply]
you can do following steps:
```
wget https://github.com/open-watcom/open-watcom-v2/releases/download/Current-build/owsnapshot.tar.xz -O /tmp
mkdir -p /usr/bin/watcom
tar -xf /tmp/backup.tar.xz -C /usr/bin/watcom
rm /tmp/backup.tar.xz
export WATCOM=/usr/bin/watcom
export PATH=$WATCOM/binl64:$WATCOM/binw:$PATH
export INCLUDE=$WATCOM/h
```
